WooCommerce/Classes/ViewModels/Order Details/Shipping Labels/BoldableTextView.swift

```diff
@@ -0,0 +1,113 @@
+import SwiftUI
+
+/// A text view that takes in a string whose substrings within two asterisks are bolded.
+///
+/// For example, initializing with a string "*Never* will I *stop learning*." will bold the substrings "Never" and "stop learning".
+///
+struct BoldableTextView: View {
+    fileprivate struct Element {
+        let content: String
+        let isBold: Bool
+
+        init(content: String, isBold: Bool) {
+            if isBold {
+                self.content = content.replacingOccurrences(of: boldRegionMark, with: "")
+            } else {
+                self.content = content
+            }
+            self.isBold = isBold
+        }
+    }
+
+    fileprivate static let boldRegionMark = "*"
+
+    private let elements: [Element]
+
+    init(_ content: String) {
+        elements = content.parseBoldableElements()
+    }
+
+    var body: some View {
+        elements.map { $0.toTextView }
+            .reduce(into: Text(""), { result, text in
+                result = result + text
+            })
+    }
+}
+
+private extension BoldableTextView.Element {
+    var toTextView: Text {
+        if isBold {
+            return Text(content).fontWeight(.bold)
+        } else {
+            return Text(content)
+        }
+    }
+}
+
+private extension String {
+    /// Parses the input text and returns a collection of boldable elements.
+    ///
+    /// - Returns: A collection of boldable elements.
+    func parseBoldableElements() -> [BoldableTextView.Element] {
+        let boldRegionMark = BoldableTextView.boldRegionMark
+        let regex = try! NSRegularExpression(pattern: "\\\(boldRegionMark){1}(.*?)\\\(boldRegionMark){1}")
+        let range = NSRange(location: 0, length: count)
```

Comment:
`String` can use different representations of the text that it stores (utf8 / utf16) and `count` may differ from `NSString` length. 

`NSRange` has an initialiser that takes range and swift string. I think it's safer to use :-)
```
extension _NSRange {
    public init<R, S>(_ region: R, in target: S) where R : RangeExpression, S : StringProtocol, R.Bound == String.Index
}
```

Reply:
I didn't think of the potential difference in `count` between `String` and `NSString`! updated here: https://github.com/woocommerce/woocommerce-ios/pull/3285/commits/b6ada23fe642cb118d1d635c280ec10d9ce79850